microsite/docusaurus.config.ts

```diff
@@ -0,0 +1,357 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// @ts-check
+
+/** @type{import('prism-react-renderer').PrismTheme} **/
+// @ts-ignore
+import { themes } from 'prism-react-renderer';
+import type * as Preset from '@docusaurus/preset-classic';
+import { Config } from '@docusaurus/types';
+const backstageTheme = themes.vsDark;
+backstageTheme.plain.backgroundColor = '#232323';
+
+const config: Config = {
+  title: 'Backstage Software Catalog and Developer Platform',
+  tagline: 'An open source framework for building developer portals',
+  url: 'https://backstage.io',
+  baseUrl: '/',
+  organizationName: 'Spotify',
+  projectName: 'backstage',
+  scripts: [
+    'https://buttons.github.io/buttons.js',
+    'https://unpkg.com/medium-zoom@1.0.6/dist/medium-zoom.min.js',
+    '/js/medium-zoom.js',
+    '/js/dismissable-banner.js',
+    '/js/scroll-nav-to-view-in-docs.js',
+  ],
+  stylesheets: [
+    'https://fonts.googleapis.com/css?family=IBM+Plex+Mono:500,700&display=swap',
+  ],
+  favicon: 'img/favicon.ico',
+  customFields: {
+    fossWebsite: 'https://spotify.github.io/',
+    repoUrl: 'https://github.com/backstage/backstage',
+  },
+  onBrokenLinks: 'log',
+  onBrokenMarkdownLinks: 'log',
+  presets: [
+    [
+      '@docusaurus/preset-classic',
+      /** @type {import('@docusaurus/preset-classic').Options} */
+      {
+        docs: {
+          editUrl: 'https://github.com/backstage/backstage/edit/master/docs/',
+          path: '../docs',
+          sidebarPath: 'sidebars.json',
+        },
+        blog: {
+          path: 'blog',
+        },
+        theme: {
+          customCss: 'src/theme/customTheme.scss',
+        },
+        gtag: {
+          trackingID: 'G-KSEVGGNCJW',
+        },
+      },
+    ],
+  ],
+  markdown: {
+    preprocessor({ fileContent }) {
+      // Replace all HTML comments with empty strings as these are not supported by MDXv2.
+      function removeHtmlComments(input) {
+        let previous;
+        do {
+          previous = input;
+          input = input.replace(/<!--|--!?>/g, '');
+        } while (input !== previous);
+        return input;
+      }
+      return removeHtmlComments(fileContent);
```

Comment:
This doesn't seem to do what we want. How come you changed it from the previous `return fileContent.replace(/<!--.*?-->/gs, '');`?

For example, look at http://localhost:3000/docs/features/techdocs/, the `Intro, backstory, etc.:` is part of a comment and should not be displayed

Reply:
because of this https://codeql.github.com/codeql-query-help/javascript/js-incomplete-multi-character-sanitization/#example

Reply:
i think this fixes this 
```
          input = input.replace(/<!--.*?-->/g, '');

```

Reply:
Ah I see, you can keep it as it was before, we're only dealing with trusted input.